cpp/cmake/thirdparty/get_arrow.cmake

```diff
@@ -80,8 +80,14 @@ function(find_and_configure_arrow VERSION BUILD_STATIC ENABLE_S3 ENABLE_ORC ENAB
     list(APPEND ARROW_PYTHON_OPTIONS "ARROW_PYTHON ON")
     # Arrow's logic to build Boost from source is busted, so we have to get it from the system.
     list(APPEND ARROW_PYTHON_OPTIONS "BOOST_SOURCE SYSTEM")
-    list(APPEND ARROW_PYTHON_OPTIONS "Thrift_SOURCE BUNDLED")
-    list(APPEND ARROW_PYTHON_OPTIONS "ARROW_DEPENDENCY_SOURCE AUTO")
```

Comment:
You need to keep the `list(APPEND ARROW_PYTHON_OPTIONS "ARROW_DEPENDENCY_SOURCE AUTO")` for python. Without I see failure due to ARROW expecting rapidjson to be installed on the system.

Reply:
Thanks I just pushed that change, please take another look